errors.go

```diff
@@ -89,14 +90,22 @@ func ErrorFilter(req Request, svc Service) Response {
 		b, _ := rsp.BodyBytes(false)
 		switch rsp.Header.Get("Terror") {
 		case "1":
+			var err error
 			tp := &terrorsproto.Error{}
-			if err := json.Unmarshal(b, tp); err != nil {
+
+			switch rsp.Header.Get("Content-Type") {
+			case "application/octet-stream", "application/x-google-protobuf", "application/protobuf":
```

Comment:
As far as I can see we never set a content type of `application/octet-stream`, is there a particular reason why we want to support it here?

Reply:
I wanted to support types commonly sent, [protorpc uses octet-stream](https://github.com/google/protorpc/commit/eb03145a6a7c72ae6cc43867d9635a5b8d8c4545#diff-9409428171f083cd5b68e3037d0f92756b895617effda1a591609e609b4c6f2eR49-R51)

I don't feel too strongly either way. I don't think this makes this path any less correct?

Reply:
Yeh, it's probably fine. I think what made me question it is thinking that it's really easy to add later but can be quite difficult to remove and unlike the others this content type isn't really linked to proto – there are lots of things that would use this content type.